# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/WasmFeatures.h
# Commit: 27db0e6a7010
# Full Hash: 27db0e6a7010f290ac14e5d260c0b4fe48442486
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 6
# Description:
#   Bug 1727084 - Memory64 - Ion operations. r=yury
#   
#   Add memory64 generalizations for Ion for all operations (except bulk
#   memory, done in an earlier patch).
#   
# ==============================================================================

diff -r 1ab8383d17ee -r 27db0e6a7010 js/public/WasmFeatures.h
--- a/js/public/WasmFeatures.h	Mon Oct 18 07:31:56 2021 +0000
+++ b/js/public/WasmFeatures.h	Mon Oct 18 07:31:56 2021 +0000
@@ -139,7 +139,8 @@
   EXPERIMENTAL(/* capitalized name   */ Memory64,                             \
                /* lower case name    */ memory64,                             \
                /* compile predicate  */ WASM_MEMORY64_ENABLED,                \
-               /* compiler predicate */ BaselineAvailable(cx),                \
+               /* compiler predicate */ BaselineAvailable(cx) ||              \
+                  IonAvailable(cx),                                           \
                /* flag predicate     */ !IsFuzzingIon(cx) &&                  \
                    !IsFuzzingCranelift(cx),                                   \
                /* shell flag         */ "memory64",                           \